src/controllers/referee.js

```diff
@@ -0,0 +1,60 @@
+import {ReturnState} from './_base.js';
+
+/**
+ * Clean the incoming POST request body to make it more compatible with the
+ * database and its validation rules.
+ *
+ * @param {object} body The incoming request's body.
+ * @returns {any} A json object that's just got our cleaned up fields on it.
+ */
+const cleanInput = (body) => {
+  return {
+    // The string is trimmed for leading and trailing whitespace and then
+    // copied across if they're in the POST body or are set to undefined if
+    // they're missing.
+    cleanRefereeName: body.refereeName === undefined ? undefined : body.refereeName.trim(),
+    cleanRefereeEmail: body.refereeEmail === undefined ? undefined : body.refereeEmail.trim()
+  };
+};
+
+const refereeController = (request) => {
+  // Clear any existing errors.
+  request.session.refereeError = false;
+
+  request.session.refereeNameError = false;
+  request.session.refereeEmailEmptyError = false;
+  request.session.refereeEmailInvalidError = false;
+
+  // Clean up the user's input before we use it.
+  const {cleanRefereeName, cleanRefereeEmail} = cleanInput(request.body);
+
+  // Save the referee details.
+  request.session.refereeName = cleanRefereeName;
+  request.session.refereeEmail = cleanRefereeEmail;
+
+  request.session.refereeNameError = cleanRefereeName === undefined || cleanRefereeName.trim() === '';
+
+  request.session.refereeEmailEmptyError = cleanRefereeEmail === undefined || cleanRefereeEmail.trim() === '';
```

Comment:
`cleanXYZ.trim()` is unnecessary here as `cleanXYZ` was already `trim`med in `cleanInput()`.

Reply:
Resolved in 835c3f6e21d690211769853ecc625e19d00292c6.